roomserver/api/query.go

```diff
@@ -255,6 +255,13 @@ type QueryRoomVersionCapabilitiesResponse struct {
 	AvailableRoomVersions map[string]string `json:"available"`
 }
 
+type QueryFullStateAtEventRequest struct {
+	EventID string `json:"event_id"`
+}
+type QueryFullStateAtEventResponse struct {
```

Comment:
As of https://github.com/matrix-org/dendrite/pull/962 you can use that query API instead of adding another.